web/apps/labelstudio/src/components/HeidiTips/utils.ts

```diff
@@ -1,16 +1,74 @@
-import { TipsCollection } from "./content";
-import type { Tip } from "./types";
+import { defaultTipsCollection } from "./content";
+import type { Tip, TipsCollection } from "./types";
 
 const STORE_KEY = "heidi_ignored_tips";
+const CACHE_KEY = "heidi_live_tips_collection";
+const CACHE_FETCHED_AT_KEY = "heidi_live_tips_collection_fetched_at";
+const CACHE_STALE_TIME = 1000 * 60 * 60; // 1 hour
+const MAX_TIMEOUT = 5000; // 5 seconds
 
 function getKey(collection: string) {
   return `${STORE_KEY}:${collection}`;
 }
 
-export function getRandomTip(collection: keyof typeof TipsCollection): Tip | null {
+export const loadLiveTipsCollection = () => {
+  // stale while revalidate
+  const cachedData = localStorage.getItem(CACHE_KEY);
+  const fetchedAt = localStorage.getItem(CACHE_FETCHED_AT_KEY);
+
+  // Read from local storage if the collection is less than CACHE_STALE_TIME milliseconds old
+  if (cachedData && fetchedAt && Date.now() - Number.parseInt(fetchedAt) < CACHE_STALE_TIME) {
+    return JSON.parse(cachedData);
+  }
+
+  const abortController = new AbortController();
+
+  // Abort the request after MAX_TIMEOUT milliseconds to ensure we don't block for too long, something might be wrong with the network
```

Comment:
```suggestion
  // Abort the request after MAX_TIMEOUT milliseconds to ensure we won't wait for too long, something might be wrong with the network or it could be an air-gapped instance
```